packages/orbit-components/src/InputField/index.tsx

```diff
@@ -433,6 +434,7 @@ const InputField = React.forwardRef<HTMLInputElement, Props>((props, ref) => {
           name={name}
           type={getDOMType(type)}
           value={value}
+          defaultValue={defaultValue}
```

Comment:
Given what you wrote above, should we alter the logic to pass either `value` or `defaultValue` and not both at the same time?

Reply:
I have had that question in my life in the past and ended up concluding that the responsibility of knowing when to use `value` + `onChange` or `defaultValue` (controlled vs uncontrolled) is the developer's.

The console will show a warning in dev mode if someone tries to use both at the same time.
<img width="498" alt="image" src="https://github.com/kiwicom/orbit/assets/6265045/6734b989-0f08-47a7-ba46-4f9e14d272c8">

So I would prefer not to add that restrictive logic to our component.

Reply:
But from the typing system point of view, it's still wrong since these are incompatible, right?

IMO, static analysis tooling should help developers not rely on runtime warnings/errors.

Reply:
I tried adding conditional types, but honestly… I couldn't. I could define the type (I had to convert some interfaces to types, but all should be fine), but then passing the props to this `Input` component started throwing me errors because of type incompatibility. I might be missing something 😓

Reply:
I wouldn't bother. ideally we'd not specify these types **at all**, we'd just extend `React.InputHTMLAttributes<HTMLInputElement>` with any custom props we have, such as `error`, and be done with it

Reply:
As agreed irl, we won't add conditional types